app/Models/Signup.php

```diff
@@ -213,17 +212,23 @@ public function getAcceptedQuantity()
     }
 
     /**
-     * Scope a query to only return signups if a user is an admin, staff, or is owner of signup.
+     * Scope a query to only return signups if a user is an admin, staff, or is owner of signup and by type (optional)
      *
+     * @param array $types
      * @return \Illuminate\Database\Eloquent\Builder
      */
-    public function scopeWithVisiblePosts($query)
+    public function scopeWithVisiblePosts($query, $types = null)
     {
-        if (! is_staff_user()) {
-            return $query->with(['posts' => function ($query) {
-                $query->where('status', 'accepted')
-                ->orWhere('northstar_id', auth()->id());
-            }]);
-        }
+        return $query->with(['visiblePosts' => function ($query) use ($types) {
+
+            if ($types) {
+                $query->whereIn('type', $types);
+            }
+
+            if (! is_staff_user()) {
+               $query->where('status', 'accepted')
+                    ->orWhere('northstar_id', auth()->id());
+            }
+        }, 'posts.tags']);
     }
 }
```

Comment:
@DFurnes I'm wondering if this isn't working because this also has a query in `visiblePosts` and it is overriding something? Do we need to have `scopeWithVisiblePosts` and `visiblePosts`? Seems like to me they're doing very similar things and might be too much to have both?

Reply:
oh wait, I remember now why! You explained this before!
> Instead of scoping the query on each individual signup here, let's use the scopeVisiblePosts scope in the controller to eager-load these before they hit the transformer.

I _think_ the queries are still somehow cancelling each other out though? I tried to removed 
```
if (! is_staff_user()) {
  $query->where('status', 'accepted')
               ->orWhere('northstar_id', auth()->id());
}
```

out of `scopeWithVisiblePosts` since it should already do that in `visiblePosts` but still not getting the correct response 🤔